crates/ruff/src/rules/flake8_bugbear/rules/assignment_to_os_environ.rs

```diff
@@ -20,13 +20,13 @@ pub fn assignment_to_os_environ(checker: &mut Checker, targets: &[Expr]) {
         return;
     }
     let target = &targets[0];
-    let ExprKind::Attribute { value, attr, .. } = &target.node else {
+    let ExprKind::Attribute(ast::ExprAttribute { value, attr, .. }) = &target.node else {
         return;
     };
-    if attr != "environ" {
+    if attr.as_str() != "environ" {
```

Comment:
```suggestion
    if attr != "environ" {
```

This change isn't necessary because `Identifier` implements `PartialEq<str>`